configure.ac

```diff
@@ -1641,23 +1641,33 @@ if test "${enable_backend_kinesis}" = "yes" -o \
         "${build_ml}" = "yes"; then
     enable_cxx_linker="yes"
 
-    # Link with libatomic when using toolchains that don't provide all the
-    # atomic ops as builtins.
+    # Try to unconditionally link with -latomic. If the compiler can satisfy
+    # all the atomic ops with builtins then, the library will be left unused.
+    # Otherwise, some ops will be covered by the compiler's intrinsics and some
+    # will be picked up by the linker from -latomic. In the later case, if
+    # -latomic is not available there will be a build failure, which would
```

Comment:
I think in future the correct way would be:
- detect if latomic is there
- detect if it is needed
if needed and not present then error in configure step with user friendly explanation to install libatomic instead of build failure